spyder/widgets/variableexplorer/dataframeeditor.py

```diff
@@ -611,28 +563,226 @@ def copy(self):
+            self.endInsertRows()
+        if self.can_fetch_more() and self.axis == 0:
+            reminder = self.total_cols - self.cols_loaded
+            items_to_fetch = min(reminder, COLS_TO_LOAD)
+            self.beginInsertColumns(QModelIndex(), self.cols_loaded,
+                                    self.cols_loaded + items_to_fetch - 1)
+            self.cols_loaded += items_to_fetch
+            self.endInsertColumns()
+
+    def sort(self, column, order=Qt.AscendingOrder):
+        """Overriding sort method."""
+        ascending = order == Qt.AscendingOrder
+        self.model.sort(self.COLUMN_INDEX, order=ascending)
+        return True
+
+    def headerData(self, section, orientation, role):
+        if role == Qt.TextAlignmentRole:
+            if orientation == Qt.Horizontal:
+                return Qt.AlignCenter | Qt.AlignBottom
+            else:
+                return Qt.AlignRight | Qt.AlignVCenter
+        if role != Qt.DisplayRole:
+            return None
+        if self.axis == 1 and self._shape[1] <= 1:
+            return None
+        orient_axis = 0 if orientation == Qt.Horizontal else 1
+        if self.model.header_shape[orient_axis] > 1:
+            header = section
+        else:
+            header = self.model.header(self.axis, section)
+            if isinstance(header, TEXT_TYPES):
+                # Get the proper encoding of the text in the header.
+                # Fixes Issue 3896
+                if not PY2:
+                    try:
+                        header = header.encode('utf-8')
+                        coding = 'utf-8-sig'
+                    except UnicodeEncodeError:
+                        coding = encoding.get_coding(header)
+                else:
+                    coding = encoding.get_coding(header)
+                return to_text_string(header, encoding=coding)
+            header = to_text_string(header)
+        return header
+
+    def data(self, index, role):
+        if not index.isValid() or \
+           index.row() >= self._shape[0] or \
+           index.column() >= self._shape[1]:
+            return None
+        row, col = ((index.row(), index.column()) if self.axis == 0
+                    else (index.column(), index.row()))
+        if role == Qt.BackgroundRole:
+            prev = self.model.header(self.axis, col - 1, row) if col else None
+            cur = self.model.header(self.axis, col, row)
+            return self._palette.midlight() if prev != cur else None
+        if role != Qt.DisplayRole:
+            return None
+        if self.axis == 0 and self._shape[0] <= 1:
+            return None
+        return to_text_string(self.model.header(self.axis, col, row))
+
+
+class DataFrameLevelModel(QAbstractTableModel):
+    """
+    Data Frame level class.
+
+    This class model the levels in the DataFrameEditor. When using multiindex
+    this model creates the labels for the index/header as Index i for each
+    section in the index/header

```

Comment:
Change this to

> This class is used to represent index levels in the DataFrameEditor. When
using MultiIndex, this model creates labels for the index/header as Index i
for each section in the index/header